sub.go

```diff
@@ -65,15 +65,15 @@ type SubscriptionCancelParams struct {
 // "now" value for billing_cycle_anchor and trial_end can be implemented
 // (they're otherwise timestamps rather than strings).
 func (p *SubscriptionParams) AppendTo(body *form.Values, keyParts []string) {
-	if p.BillingCycleAnchorNow != nil {
+	if BoolValue(p.BillingCycleAnchorNow) {
 		body.Add(form.FormatKey(append(keyParts, "billing_cycle_anchor")), "now")
 	}
 
-	if p.BillingCycleAnchorUnchanged != nil {
+	if BoolValue(p.BillingCycleAnchorUnchanged) {
 		body.Add(form.FormatKey(append(keyParts, "billing_cycle_anchor")), "unchanged")
 	}
 
-	if p.TrialEndNow != nil {
+	if BoolValue(p.TrialEndNow) {
```

Comment:
@brandur-stripe I think I missed this when I changed the AppendTo. Is this the correct way to do it? Those are custom values and we only send the custom string if the boolean is true I think. If so, we likely need to review all the AppendTo for similar mistakes.

Reply:
I don't really think it's great, but I don't have a better way of doing it right now.

Because TrialEnd is normally an integer, the only way we could have it allow the value `now` is to make it something like an `interface{}` which isn't great.

Reply:
@brandur-stripe  would you be able to go through each AppendTo and make sure I did not miss any edge-case with those if?

Reply:
Sure!

I went through and these are the ones that I found that have a similar pattern:

* `UpToInf` on `PlanTierParams`.
* `BillingCycleAnchorNow` on `SubscriptionParams`.
* `BillingCycleAnchorUnchanged` on `SubscriptionParams`.
* `DelayDaysMinimum` on `PayoutScheduleParams`.

I guess one possibility is we could have a "special" field for these that would allow you to pass a special value for these types of fields:

``` go
PlanTierParams {
    UpToSpecial: stripe.String(stripe.UpToMinimum),
}
```

You could also call this `UpToString` or `UpToEnum`.

IMO though, this isn't too much better or worse than what's already implemented, so I'd probably just leave it. What do you think?